src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -62,95 +64,125 @@ public bool HasResponseStarted
             get { return _responseStarted; }
         }
 
-        public void Consume()
+        public void Reset()
         {
-            var input = SocketInput;
-            for (; ;)
+            _onStarting = null;
+            _onCompleted = null;
+
+            _responseStarted = false;
+            _keepAlive = false;
+            _autoChunk = false;
+
+            _requestHeaders.Reset();
+            ResetResponseHeaders();
+
+            Method = null;
+            RequestUri = null;
+            Path = null;
+            QueryString = null;
+            HttpVersion = null;
+            RequestHeaders = _requestHeaders;
+            MessageBody = null;
+            RequestBody = null;
+            StatusCode = 200;
+            ReasonPhrase = null;
+            ResponseHeaders = _responseHeaders;
+            ResponseBody = null;
+            DuplexStream = null;
+
+        }
+
+        public void ResetResponseHeaders()
+        {
+            _responseHeaders.Reset();
+            _responseHeaders.HeaderServer = "Kestrel";
+            _responseHeaders.HeaderDate = DateTime.UtcNow.ToString("r");
+        }
+
+        public async void ProcessFraming()
```

Comment:
That's inconvenient... I fixed it with `Task.Run((Action)_frame.ProcessFraming);` Dunno why it couldn't take an action automatically, but it could take the func.